### PR TITLE
Convert DW drivers to DT_INST

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: boards/arc/hsdk/hsdk.dts [hunk 1]
```diff
@@ -46,4 +46,13 @@
 &spi2 {
 	status = "okay";
 	clock-frequency = <33333333>;
+
+	cs-gpios = <&gpio0 9 0>;
+	ili9340@0 {
+		compatible = "ilitek,ili9340";
```
Comment (ruuddw) [inline boards/arc/hsdk/hsdk.dts]: I believe this would be an add-on board/shield used for testing and demo's, but not standard part of the hsdk board.
Should it be here, or in some overlay/test/application only?
This seems unrelated to the rest of this PR.

Comment (galak) [reply boards/arc/hsdk/hsdk.dts]: So this commit added support at the SoC level for this display:

commit fd17b86b946bc21a790ca3f4306531c7e713efd8
Author: Watson Zeng <zhiwei@synopsys.com>
Date:   Tue Sep 10 14:45:29 2019 +0800

    arc: hsdk: add lvgl support for hsdk board

I'm not sure about if this should be a shield or what, but this should have been done as a dts change and not as a set of #defines in `soc/arc/snps_arc_hsdk/dts_fixup.h`.  The reason the change is in this PR is because the designware driver changes and up impacting those defines that should be generated by DTS.  Thus I moved it to DTS.

Comment (ruuddw) [reply boards/arc/hsdk/hsdk.dts]: Thanks, I missed that the defines for this were already in the dts_fixup and wondered where they came from.
I think this can use a clean-up to avoid confusion, but that can be a different PR. I'll check with Watson.

**Review by ruuddw — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]
LGTM USB part

File: boards/arc/hsdk/hsdk.dts (hunk 1 above)
Comment (galak) [reply boards/arc/hsdk/hsdk.dts]: Thanks.
